frontend/src/App.js

```diff
@@ -31,6 +32,7 @@ function App() {
             <Route path="/home" element={<PrivateRoute path="/home" />}>

```

Comment:
Hi, I just notice but you do not need the "path" prop for <Route />
`<Route element={<PrivateRoute path="/home" />}>` will work

Reply:
Not really right?
```
useEffect(() => {
    const res = axios
      .get(props.path, { withCredentials: true })
      .catch((error) => {
        console.log("Error!");
      });
    if (res && res.status === STATUS_CODE_OK) {
      setAuthentication(true);
    } else {
      setAuthentication(false);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []); // Will run once when component first mounts
```
Because in the axios get(props.path..., I am trying to check if I am able to get the props.path page. If there is no response from the axios, it means the user is not authenticated to access this particular path, thus setAuthentication(false), else setAuthentication(true).

Reply:
Ermm actually that's not how axios.get work, `axios.get(URL, options)` takes `URL` which is actually the endpoint URL of the backend, not the URL of the frontend. In fact, we do not need to pass the frontend URL to the backend since we only need to check if the user has the valid token.

The flow is in the following order:
1. private route send GET request along with cookie to user-service 
2. user service check if this token is valid and send a response with status to the frontend 
3. frontend receive the response status from backend and check if it's 200, if yes, redirect to private route, else redirect to signin page

In this flow, you see that we don't need to pass the frontend URL, since we are not checking user access level based on the URL such that different access only access certain URL. In our case, there is only 1 access level, that is, if you have a valid token, then you will be able to access all the private route.

I have tested this code too, although it works to prevent the user without valid token to access the route, it does not let the user go to that route when the user has the valid token (after signin).

I have helped to write the code for this part, you can check this file in my PR.

Reply:
Oh I understand now. Thanks for the clarification! 👍